### ISSUE TITLE
Wrong column reported in no-missing-end-of-source-newline

### ISSUE THREAD

**Issue by Arcanemagus** [stylelint/stylelint#1892]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where `no-missing-end-of-source-newline` returns an invalid column for a SugarSS file.

> Which rule, if any, is this issue related to?

`no-missing-end-of-source-newline`

> What CSS is needed to reproduce this issue?

```sugarss
body
```

> What stylelint configuration is needed to reproduce this issue?

_stylelint.config.js_:

```js
module.exports = {
  "rules": {
    "no-missing-end-of-source-newline": true
  }
}
```

Note, example files can be found here: [no-missing-end-of-source-newline_linter-stylelint_GH264.zip](https://github.com/stylelint/stylelint/files/462741/no-missing-end-of-source-newline.zip)

> Which version of stylelint are you using?

`7.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint foo.sss --syntax sugarss`, and the Node.js API (Atom `linter-stylelint` package).

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SugarSS.

> What did you expect to happen?

The proper column (4?) to be returned.

> What actually happened (e.g. what warnings or errors you are getting)?

Stylelint reports the message as being on column 7, of a file that only has 4 characters _total_.

![image](https://cloud.githubusercontent.com/assets/427137/18367551/90744f28-75d0-11e6-9e56-00b549993ca8.png)

Originally filed as part of https://github.com/AtomLinter/linter-stylelint/issues/264.

**Comment by Arcanemagus** [comment]
Looks like this might be "always 2 characters more", the second error in the GIF in the linked bug report is also this bug, but instead it's reporting an error on `3:1` on a file where `2:12` is the last possible column.

**Comment by evilebottnawi** [comment]
@ai Strange behaviour
My code
```shell
a
  color: pink

b
  color: orange
```js
const postcss = require('postcss');

module.exports = postcss.plugin('PLUGIN_NAME', (opts) => {
    opts = opts || {};

    return (root, result) => {
        const sourceCss = root.source.input.css
        console.log(root.positionBy({ index: sourceCss.length - 1 }))
    };
});
```
Result: `{ line: 6, column: 9 }`
Though actually be `{ line: 5, column: 15 }`. Maybe related to `sugarss`.

**Comment by jeddy3** [comment]
Can you raise this in the SugarSS issue tracker?

**Comment by ai** [comment]
Does this issue is related? https://github.com/postcss/sugarss/issues/49

**Comment by jeddy3** [comment]
Looks like it!

**Comment by ai** [comment]
@jeddy3 :D I reported a issue to other project, but this error happens because of my project :). Karma :).

I will fix it.

What do you think about more bullet-proof Stylelint plugins? I think it is wierd, that Stylelint fails on wrong source maps. Honestly, source maps is tricky thing and I am sure that there are many mistakes in other source map implementations.

**Comment by davidtheclark** [comment]
> What do you think about more bullet-proof Stylelint plugins?

@Arcanemagus @ai Yeah ... so there will be no problem as long as the mistaken column is not past the end of the line. If it's at the end of the line, the plugin breaks, huh. Is there no way around that? 

Not that we shouldn't fix the issues that come up — just hoping such an insignificant issue doesn't have to break UX.

**Comment by Arcanemagus** [comment]
Throttling of the display of the error definitely needs to be added, it's just not something I've dedicated time to yet.

**Comment by ai** [comment]
Fixed in SugarSS 0.2. Thanks for helping in investigation.

**Comment by evilebottnawi** [comment]
@ai Thanks, in the near future it'll do.

**Comment by evilebottnawi** [comment]
Done in https://github.com/stylelint/stylelint/pull/1976
